### PR TITLE
Add hover to clipboard

### PR DESCRIPTION
**Pull request by pivilartisant**
[Screencast from 04-07-2023 17:50:18.webm](https://github.com/massalabs/ui-kit/assets/90157528/21fccc7e-02fd-4933-917f-c7d037ccc8aa)

BTW : you don't see the mouse on video but it is here ^^

By default it is true. We don't need to modify the code in our existing code base to make the feature functional. 
Possible to disable hover using the following param : 

``` 
toggleHover={false}
```

### CODE REVIEW

**Review by Pierregvx — APPROVED** [review]

File: src/components/Clipboard/Clipboard.tsx [hunk 1]
```diff
@@ -50,12 +56,42 @@ export function Clipboard(props: ClipboardProps) {
     }, 3000);
   }
 
+  function handleMouseMove(event: React.MouseEvent<HTMLDivElement>) {
+    setPosition({ x: event.clientX, y: event.clientY });
+  }
```
Comment (maxime-carabina) [inline src/components/Clipboard/Clipboard.tsx:61]:
```suggestion
```

File: src/components/Clipboard/Clipboard.tsx [hunk 2]
```diff
@@ -50,12 +56,42 @@ export function Clipboard(props: ClipboardProps) {
...
   return (
     <>
+      {isHovered && hover ? renderAddressHover() : null}
       <div
         data-testid="clipboard-field"
         className={`flex flex-row items-center mas-body2 justify-between w-full
         h-12 px-3 rounded bg-secondary cursor-pointer ${customClass}`}
+        onMouseEnter={() => handleHover(true)}
+        onMouseLeave={() => handleHover(false)}
+        onMouseMove={handleMouseMove}
```
Comment (maxime-carabina) [inline src/components/Clipboard/Clipboard.tsx:94]:
```suggestion
        onMouseMove={(e) => setPosition({ x: event.clientX, y: event.clientY })}
```

File: src/components/Clipboard/Clipboard.tsx [hunk 3]
```diff
@@ -50,12 +56,42 @@ export function Clipboard(props: ClipboardProps) {
...
+  function renderAddressHover() {
+    return (
+      <p
+        style={divStyle}
+        className="relative bg-secondary z-10 top-2 text-f-primary pr-2 w-fit truncate pointer-events-none px-1 rounded"
+      >
+        {rawContent}
+      </p>
+    );
+  }
```
Comment (maxime-carabina) [inline src/components/Clipboard/Clipboard.tsx:78]:
```suggestion
  function displayAddressOnHover() {
    const divStyle: CSSProperties = {
      position: 'fixed',
      left: position.x + 30,
      top: position.y - 30,
    };
  
    return (
      <div
        style={divStyle}
        className="relative bg-secondary z-10 top-2 text-f-primary pr-2 w-fit truncate pointer-events-none px-1 rounded"
      >
        {rawContent}
      </div>
    );
  }
```

If `divStyle` is only use in `displayAddressOnHover`, you can move this `const` into the function. 🤔

Comment (pivilartisant) [reply src/components/Clipboard/Clipboard.tsx:78]: thanks for that !

Comment (maxime-carabina) [reply src/components/Clipboard/Clipboard.tsx]: @pivilartisant Can you rename the function as I wrote above? It's to make it more explicit 😉

Comment (pivilartisant) [reply src/components/Clipboard/Clipboard.tsx]: ouaip j'avais pas vu

**Review by maxime-carabina — APPROVED** [review]
LGTM 🎉

**Pull request merged**